jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -454,16 +498,28 @@ public JibContainerBuilder toContainerBuilder() throws IOException {
     // Add others to layer configuration
     for (Path path : addedOthers) {
       if (Files.isDirectory(path)) {
-        layerConfigurationsBuilder.addDirectoryContents(
-            LayerType.EXTRA_FILES, path, path1 -> true, appRoot.resolve(othersDestination));
+        addDirectoryContentsToLayer(
+            layerBuilders,
+            LayerType.EXTRA_FILES,
+            path,
+            path1 -> true,
+            appRoot.resolve(othersDestination));
       } else {
-        layerConfigurationsBuilder.addFile(
+        addFileToLayer(
+            layerBuilders,
             LayerType.EXTRA_FILES,
             path,
             appRoot.resolve(othersDestination).resolve(path.getFileName()));
       }
     }
-    jibContainerBuilder.setLayers(layerConfigurationsBuilder.build().getLayerConfigurations());
+
+    // Add layer configurations to container builder
+    ImmutableMap.Builder<LayerType, LayerConfiguration> layerConfigurationsMap =
```

Comment:
We no longer need a map, so probably this can be
```java
      List<LayerConfiguration> layers = new ArrayList<>();
      layerBuilders.forEach(
          (type, builder) -> layers.add(builder.setName(type.getName()).build()));
```
I think we live without `ImmutableList`.